client/lib/protect-form/index.jsx

```diff
@@ -14,37 +14,58 @@ const debug = debugModule( 'calypso:protect-form' );
 const confirmText = i18n.translate( 'You have unsaved changes. Are you sure you want to leave this page?' );
 const beforeUnloadText = i18n.translate( 'You have unsaved changes.' );
 let formsChanged = [];
+let listenerCount = 0;
 
+function warnIfChanged( event ) {
+	if ( ! formsChanged.length ) {
+		return;
+	}
+	debug( 'unsaved form changes detected' );
+	( event || window.event ).returnValue = beforeUnloadText;
+	return beforeUnloadText;
+}
+
+function addBeforeUnloadListener() {
+	if ( listenerCount === 0 ) {
+		window.addEventListener( 'beforeunload', warnIfChanged );
```

Comment:
This is minor but I'd recommend checking for window before we call it. If we ever want to test this, it can be a pain to work around.